nf_core/pipeline-template/pyproject.toml

```diff
@@ -1,10 +1,14 @@
-# Config file for Python. Mostly used to configure linting of bin/check_samplesheet.py with Black.
+# Config file for Python. Mostly used to configure linting of bin/check_samplesheet.py with Ruff.
 # Should be kept the same as nf-core/tools to avoid fighting with template synchronisation.
-[tool.black]
+[tool.ruff]
 line-length = 120
-target_version = ["py37", "py38", "py39", "py310"]
+target-version = "py38"
+ignore = ["E501"]
+select = ["I", "E4", "E7", "E9", "F"]
```

Comment:
I think "E1" and "E2" would also make sense to have. Or would you like to do that in a separate PR, because it might include more changes (we can ignore that specific commit with ` --ignore-rev`)

Reply:
["N"](https://docs.astral.sh/ruff/rules/#pep8-naming-n) for keeping variable and function names in line wuld be nice as well.

Reply:
I have added `E1`, `N` and `UP` to this PR, the changes were easy. `E2` is still unstable, and we see the message `warning: Selection E2 has no effect because the --preview flag was not included.`, so it will be better to add in once it is stable.